references/obj_detection/train_pytorch.py

```diff
@@ -21,32 +22,57 @@
 
 from doctr.datasets import DocArtefacts
 from doctr.utils import DetectionMetric
+from doctr.transforms.functional.pytorch import rotate
 
 
-def convert_to_abs_coords(targets, img_shape):
+def convert_to_abs_coords(targets, img_shape, is_transform=False, **kwargs):
     height, width = img_shape[-2:]
-
     for idx in range(len(targets)):
         targets[idx]['boxes'][:, 0::2] = (targets[idx]['boxes'][:, 0::2] * width).round()
         targets[idx]['boxes'][:, 1::2] = (targets[idx]['boxes'][:, 1::2] * height).round()
-
-    targets = [{
-        "boxes": torch.from_numpy(t['boxes']).to(dtype=torch.float32),
-        "labels": torch.tensor(t['labels']).to(dtype=torch.long)}
-        for t in targets
-    ]
-
-    return targets
+    if is_transform:
+        if bool(random.getrandbits(1)):
+            angle = random.uniform(0.1, 5.0)
+        else:
+            angle = random.uniform(-0.1, -5.0)
+        bbox = np.array([i["boxes"] for i in targets], dtype=object)
+        images = kwargs.get('images',)
+        for id in range(len(bbox)):
+            images[id], bbox[id] = rotate(images[id], bbox[id], angle)
+            bbox_t = bbox[id][:, :4]
+            for z in range(len(bbox[id])):
+                bbox_t[z][0] = bbox[id][z][0] - bbox[id][z][2] / 2
+                bbox_t[z][1] = bbox[id][z][1] - bbox[id][z][3] / 2
+                bbox_t[z][2] = bbox_t[z][0] + bbox[id][z][2]
+                bbox_t[z][3] = bbox_t[z][1] + bbox[id][z][3]
+            bbox[id] = bbox_t
+        targets = [{
+            "boxes": torch.from_numpy(bo).to(dtype=torch.float32),
+            "labels": torch.tensor(t['labels']).to(dtype=torch.long)}
+            for bo, t in zip(bbox, targets)
+        ]
+        return images, targets
+    else:
+        targets = [{
+            "boxes": torch.from_numpy(t['boxes']).to(dtype=torch.float32),
+            "labels": torch.tensor(t['labels']).to(dtype=torch.long)}
+            for t in targets
+        ]
+        return targets
 
 
-def fit_one_epoch(model, train_loader, optimizer, scheduler, mb, ):
+def fit_one_epoch(model, train_loader, optimizer, scheduler, mb, is_transform=True, ):
     model.train()
     train_iter = iter(train_loader)
     # Iterate over the batches of the dataset
     for _ in progress_bar(range(len(train_loader)), parent=mb):
         images, targets = next(train_iter)
         optimizer.zero_grad()
-        targets = convert_to_abs_coords(targets, images.shape)
+        if is_transform:
+            kwargs = {"images": images}
+            images, targets = convert_to_abs_coords(targets, images.shape, is_transform=True, **kwargs)
+        else:
+            targets = convert_to_abs_coords(targets, images.shape)
```

Comment:
Let's make augmentation separate :ok_hand: 
This is called "convert_to_abs_coords" so it's expected to do what it says it does and being deterministic

You can add a `data_augmentation` function that takes images + targets, and returns both, after the conversion :+1:

Reply:
Thanks, alright. I have added the function named as data_augmentations. However, since I think, before converting to tensor the values of the "targets" in covert_to_abs_coords, would it not be better if instead we call the functions data_augmentation inside it, and then at the end we can convert it all to tensor as required?